quill-core/src/test/scala/io/getquill/util/LogToFileSpec.scala

```diff
@@ -0,0 +1,24 @@
+package io.getquill.util
+
+import io.getquill.testContext._
+import io.getquill.{ Spec, testContext }
+
+import scala.io.Source
+
+class LogToFileSpec extends Spec {
+
+  "logs a query to file when enabled" in {
+    val q = quote {
+      qr1.leftJoin(qr2).on((a, b) => a.i == b.i)
+    }
+    testContext.run(q)
+
+    val queryFile = Source.fromFile("queries.sql")
+    val contents = queryFile.mkString.trim
+    queryFile.close()
+
+    contents must not be empty
+    contents must endWith("""querySchema("TestEntity").leftJoin(querySchema("TestEntity2")).on((a, b) => a.i == b.i).map(x => (x._1.s, x._1.i, x._1.l, x._1.o, x._1.b, x._2.map((v) => v.s), x._2.map((v) => v.i), x._2.map((v) => v.l), x._2.map((v) => v.o)));""")
```

Comment:
The file queries.sql is made during compile time, this test is going to run only after `qr1.leftJoin(qr2).on((a, b) => a.i == b.i)` and all the other queries in the whole build are compiled (hence are already in queries.sql) so you can't check the last line so you'd have to search the entire thing.

I think an easier way to test this stuff is to just abstract his logic out to that function:
```scala
def doLog(sourcePath: String, line: Int, column: Int)
      quillLogFile match {
        case LogToFile.Enabled(_) =>
          runtime.unsafeRunAsync_(log.info(
            s"""
                 |-- file: ${sourcePath}
                 |-- line: ${line}
                 |-- col: ${column}
                 |-- time: ${ZonedDateTime.now().format(LogDatetimeFormatter.isoLocalDateTimeFormatter)}
                 |${idiom.format(queryString)};
                 |""".stripMargin
          ))
```
The just call that manually from the unit test. That way the query should actually be the last thing in the log.
Be sure to read just the last line of this log file because it can get _really_ big.
Here's how you could do that:
https://stackoverflow.com/a/44933786

Reply:
Addressed this by extracting this to a class as suggested.  Makes the unit test a lot simpler!